File: old/basic/src/main/java/com/dinesh/android/rv/java/search/RvMain.java
package com.dinesh.android.rv.java.search;

import android.os.Bundle;
import android.util.Log;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.SearchView;
import androidx.recyclerview.widget.DividerItemDecoration;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.dinesh.android.R;
import com.dinesh.android.app.ToolbarMain;

import java.util.ArrayList;
import java.util.List;
import java.util.Locale;

public class RvMain extends ToolbarMain implements RvInterface {
    private final String TAG = "log_" + RvMain.class.getName().split(RvMain.class.getName().split("\\.")[2] + ".")[1];

    List<RvModel> rvModelList = new ArrayList<>();
    RvAdapter rvAdapter;
    RecyclerView recyclerView;
    SearchView searchView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentViewLayout(R.layout.rv_basic_main);

        recyclerView = findViewById(R.id.recyclerView);
        searchView = findViewById(R.id.searchView);

        addDataToList();

        recyclerView.setHasFixedSize(true);
        rvAdapter = new RvAdapter(rvModelList, RvMain.this);
        recyclerView.setLayoutManager(new LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false));
        recyclerView.setAdapter(rvAdapter);
        recyclerView.addItemDecoration(new DividerItemDecoration(this, DividerItemDecoration.VERTICAL));

        searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
            @Override
            public boolean onQueryTextSubmit(String query) {
                return false;
            }

            @Override
            public boolean onQueryTextChange(String newText) {
                filterList(newText);
                return true;
            }
        });
    }

    private void filterList(String query) {
        if (query != null) {
            ArrayList<RvModel> filteredList = new ArrayList<>();
            for (RvModel i : rvModelList) {
                if (i.name.toLowerCase(Locale.ROOT).contains(query.toLowerCase(Locale.ROOT))
                        || String.valueOf(i.profilePic).toLowerCase(Locale.ROOT).contains(query.toLowerCase(Locale.ROOT))) {
                    filteredList.add(i);
                }
            }

            if (filteredList.isEmpty()) {
                Log.e(TAG, "filterList: No Data found");
                rvAdapter.setFilteredList(new ArrayList<>());
            } else {
                rvAdapter.setFilteredList(filteredList);
            }
        }
    }

    private void addDataToList() {
        // Sample Model Data
        for (int i = 0; i <= 50; i++) {
            rvModelList.add(new RvModel(R.drawable.ic_launcher_foreground, "User " + (i + 1), false, false));
        }
    }

    @Override
    public void onItemClick(View view, int position) {
        Log.i(TAG, "onItemClick: " + position);
    }
}

File: old/basic/src/main/java/com/dinesh/android/rv/kotlin/search/RvMain.kt
package com.dinesh.android.rv.kotlin.search

import android.view.View
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.dinesh.android.R
import android.util.Log
import android.widget.Toast
import androidx.appcompat.widget.SearchView
import androidx.recyclerview.widget.DividerItemDecoration
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.dinesh.android.app.ToolbarMain
import java.util.*
import kotlin.collections.ArrayList

class RvMain : ToolbarMain(), RvInterface {
    private val TAG = "log_" + RvMain::class.java.name.split(RvMain::class.java.name.split(".").toTypedArray()[2] + ".").toTypedArray()[1]

    private lateinit var recyclerView: RecyclerView
    private lateinit var searchView: SearchView
    private var rvModelList = ArrayList<RvModel>()
    private lateinit var rvAdapter: RvAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentViewLayout(R.layout.rv_basic_main)

        recyclerView = findViewById(R.id.recyclerView)
        searchView = findViewById(R.id.searchView)

        addDataToList()

        recyclerView.setHasFixedSize(true)
        rvAdapter = RvAdapter(rvModelList, this@RvMain)
        recyclerView.layoutManager = LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false)
        recyclerView.adapter = rvAdapter
        recyclerView.addItemDecoration(DividerItemDecoration(this, DividerItemDecoration.VERTICAL))


        searchView.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
            override fun onQueryTextSubmit(query: String?): Boolean {
                return false
            }

            override fun onQueryTextChange(newText: String?): Boolean {
                filterList(newText)
                return true
            }
        })
    }

    private fun filterList(query: String?) {
        if (query != null) {
            val filteredList = ArrayList<RvModel>()
            for (i in rvModelList) {
                if (i.name.lowercase(Locale.ROOT).contains(query, ignoreCase = true) || i.profilePic.toString().lowercase(Locale.ROOT).contains(query, ignoreCase = true)) {
                    filteredList.add(i)
                }
            }

            if (filteredList.isEmpty()) {
                Log.e(TAG, "filterList: No Data found")
                rvAdapter.setFilteredList(emptyList())
            } else {
                rvAdapter.setFilteredList(filteredList)
            }
        }
    }

    private fun addDataToList() {
        // Sample Model Data
        for (i in 0..50) {
            rvModelList.add(RvModel("User " + (i + 1), R.drawable.ic_launcher_foreground))
        }
    }

    override fun onItemClick(view: View?, position: Int) {
        Log.i(TAG, "onItemClick: ${position}")
    }
}